simulation/tests/test_costs.py

```diff
@@ -210,10 +215,20 @@ def test_boiler_upgrade_scheme_grant_is_zero_when_grant_cap_exceeded(
         model = model_factory(
             start_datetime=datetime.datetime(2023, 1, 1, 0, 0),
         )
-        model.boiler_upgrade_scheme_cumulative_spend_gbp = 3_000_000
+
```

Comment:
I think to make this work with the `@property` you would need to add the following line here:
```
    model.add_agents([Household()] * num_households)
```
I think you used `24_600` in the factory, so you could define `num_households=24_600` in the test here and do it this way - WDYT?

Reply:
I think I'd need to update a number of tests such that the model has a non-empty set of household agents :/